### PR TITLE
Adding Integration tests -[2nd Set of Tests]

### PR DESCRIPTION
**Pull request by Chamindu36**
## Purpose
Adding Integration tests to apictl

## Goals
Automating Integration tests

### Covered Automated Tests from this PR
**Tests Related to APIs**

1. Export Apis with export-apis Command 

**Tests Related to Dev-first Approach**

1. Export an API with documentation
2. Export an API with a thumbnail (image)
3. Export an API with a thumbnail (on)
4. Import an API with documentation
5. Import an API with a thumbnail (image)
6. Import an API with a thumbnail (icon)
7. Check documentation updates of an API
8. Check thumbnail image update of an API
9. Check thumbnail icon update of an API
10. Initialize an API the definition.yaml (apictl init SampleAPI --definition definition.yaml)

**Tests Related to Applications**

1. List all the applications in an environment (by specifying the owner)

## Test environment
OS - Ubuntu 20.04 LTS
Java - JDK 1.8_252
APIM - 3.2.0
GO - 1.14

### CODE REVIEW

File: import-export-cli/integration/base/helper.go [hunk 1]
```diff
@@ -292,3 +294,35 @@ func CountFiles(path string) (int, error) {
 	}
 	return i, nil
 }
+
+// fileExists checks if a file exists and is not a directory before we
```
Comment (wasuradananjith) [inline import-export-cli/integration/base/helper.go:298]: Shall we change the **fileExists** to **IsFileAvailable** here?

File: import-export-cli/integration/app_test.go [hunk 2]
```diff
@@ -81,6 +81,23 @@ func TestListAppsDevopsTenantUser(t *testing.T) {
...
+	apim := apimClients[0]
+
+	for appCount := 0; appCount < 5; appCount++ {
+		testutils.AddApp(t, apim, username, password)
+	}
+
+	base.SetupEnv(t, apim.GetEnvName(), apim.GetApimURL(), apim.GetTokenURL())
+	base.Login(t, apim.GetEnvName(), username, password)
+	testutils.ValidateListAppsWithOwner(t, apim.GetEnvName())
+
```
Comment (wasuradananjith) [inline import-export-cli/integration/app_test.go:98]: Shall we remove this extra lines by the end of the function? (There are several places like this)

File: import-export-cli/integration/devFirst_test.go [hunk 3]
```diff
@@ -40,6 +43,25 @@ func TestInitializeProject(t *testing.T) {
 
 }
 
+//Initialize an API with --definition flag and import that project
+func TestInitializeAPIWithDefinitionFlag(t *testing.T) {
+	apim := apimClients[0]
+	projectName := "test"
```
Comment (wasuradananjith) [inline import-export-cli/integration/devFirst_test.go:49]: Shall we have random strings generated for the project names? @uvindra WDYT?

File: import-export-cli/integration/devFirst_test.go [hunk 4]
```diff
@@ -182,3 +204,154 @@ func TestImportProjectCreatedPassWhenAPIIsExisted(t *testing.T) {
...
+	//Update icon file to created project
+	srcPathForIcon, _ := filepath.Abs(utils.TestCase2PngPath)
+	destPathForIcon := projectPath + utils.TestCase2DestPngPathSuffix
+	base.Copy(srcPathForIcon, destPathForIcon)
+
+	time.Sleep(1 * time.Second)
+	//Import the project with updated Document and updated image thumbnail
+	testutils.ValidateImportUpdatePassedWithInitializedProject(t, args)
+
+	//Validate that image is been updated
```
Comment (wasuradananjith) [inline import-export-cli/integration/devFirst_test.go:351]: Shall we change this "is been" to "has been"? There are several comments with grammar mistakes. Can you please correct those? (ex: "a image" -> "an image")

File: import-export-cli/integration/devFirst_test.go [hunk 5]
```diff
@@ -40,6 +43,25 @@ func TestInitializeProject(t *testing.T) {
...
+	args := &testutils.InitTestArgs{
+		CtlUser:   testutils.Credentials{Username: username, Password: password},
+		SrcAPIM:   apim,
+		InitFlag:  projectName,
+		OasFlag:   utils.TestApiDefinitionPath,
+		ForceFlag: false,
+	}
+
+	testutils.ValidateInitializeProjectWithDefinitionFlag(t, args)
+
```
Comment (Meruja) [inline import-export-cli/integration/devFirst_test.go:62]: Shall we remove this extra line?

File: import-export-cli/integration/testutils/api_testUtils.go [hunk 6]
```diff
@@ -161,6 +161,25 @@ func exportAPI(t *testing.T, name string, version string, provider string, env s
 	return output, err
 }
 
+func ValidateAllApisOfATenantIsExported(t *testing.T, args *ApiImportExportTestArgs, apisAdded int) {
+	time.Sleep(5 * time.Second)
+
+	output, error := ExportAllApisOfATenant(t, args)
+	assert.Nil(t, error, "Error while Exporting APIs")
```
Comment (Meruja) [inline import-export-cli/integration/testutils/api_testUtils.go:168]: Shall we change the message like "Error while exporting APIs" in lines 168, 169 & 174?

File: import-export-cli/integration/devFirst_test.go (hunk 3 above)
Comment (uvindra) [reply import-export-cli/integration/devFirst_test.go]: +1, the rationale for this is that we can differentiate a given project from another that maybe created by a different test during a test run. This is helpful when we need to debug why a given test is failing.

Comment (Chamindu36) [reply import-export-cli/integration/devFirst_test.go]: @uvindra Fixed as you suggested

File: import-export-cli/integration/devFirst_test.go (hunk 4 above)
Comment (Chamindu36) [reply import-export-cli/integration/devFirst_test.go]: Fixed

File: import-export-cli/integration/testutils/api_testUtils.go (hunk 6 above)
Comment (Chamindu36) [reply import-export-cli/integration/testutils/api_testUtils.go]: Fixed

File: import-export-cli/integration/devFirst_test.go (hunk 5 above)
Comment (Chamindu36) [reply import-export-cli/integration/devFirst_test.go]: Fixed

File: import-export-cli/integration/base/helper.go [hunk 7]
```diff
@@ -292,3 +295,46 @@ func CountFiles(path string) (int, error) {
...
+		fmt.Println(err)
+		return err
+	}
+	defer out.Close()
+
+	_, err = io.Copy(out, in)
+	if err != nil {
+		return err
+	}
+	return out.Close()
```
Comment (uvindra) [inline import-export-cli/integration/base/helper.go:328]: Since you already have `defer out.Close()` no need to call `out.Close()` at the end

Comment (Chamindu36) [reply import-export-cli/integration/base/helper.go:328]: Fixed with f75a26712774c479a96a139dc5fdc95d089bb104

**Pull request merged**